obspy/io/gse2/core.py

```diff
@@ -6,27 +6,27 @@
 
 from obspy import Stream, Trace
 from . import libgse1, libgse2
+from obspy.core.util import open_bytes_stream
 
 
-def _is_gse2(filename):
+def _is_gse2(file):
     """
     Checks whether a file is GSE2 or not.
 
-    :type filename: str
-    :param filename: GSE2 file to be checked.
+    :type file: str or file-like object
+    :param file: GSE2 file to be checked.
     :rtype: bool
     :return: ``True`` if a GSE2 file.
     """
-    # Open file.
-    try:
-        with open(filename, 'rb') as f:
-            libgse2.is_gse2(f)
-    except Exception:
-        return False
+    with open_bytes_stream(file) as stream:
```

Comment:
In general we exclusively use that variable name "stream" only for `Stream` objects, but it's pretty clear what is up here, so I dont think we need to be too picky here, just as a side note

Reply:
Changed all occurrences of `as stream:` into `as io_stream:` (3 in total, not a big deal)